content/en/blog/_posts/2021-11-18-honor-pv-reclaim-policy.md

```diff
@@ -0,0 +1,144 @@
+
+## What is the existing behavior?
+
+Normally, if the volume is to be deleted, then the expectation is to delete the PVC for a bound PV-PVC pair. However, there is no restriction to delete a PV prior to deleting a PVC.
+
+The steps below demonstrate the current behavior
+
+#### Create PVC and wait for it to get bound.
+
+```shell
+# kubectl create -f pvc.yaml 
+persistentvolumeclaim/example-vanilla-block-pvc created
+```
+
+```shell
+# kubectl get pvc
+NAME                        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS               AGE
+example-vanilla-block-pvc   Bound    pvc-6791fdd4-5fad-438e-a7fb-16410363e3da   5Gi        RWO            example-vanilla-block-sc   19s
+```
+
+#### Delete PV.
+It can be observed that deleting a bound PV will not return back control to the shell.
+
+```shell
+# kubectl delete pv pvc-6791fdd4-5fad-438e-a7fb-16410363e3da
+persistentvolume "pvc-6791fdd4-5fad-438e-a7fb-16410363e3da" deleted
+^C
+```
+
+The PV moves into terminating state
+
+```shell
+# kubectl get pv pvc-6791fdd4-5fad-438e-a7fb-16410363e3da
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS        CLAIM                               STORAGECLASS               REASON   AGE
+pvc-6791fdd4-5fad-438e-a7fb-16410363e3da   5Gi        RWO            Delete           Terminating   default/example-vanilla-block-pvc   example-vanilla-block-sc            2m23s
+```
+
+#### Delete PVC
+
+```shell
+# kubectl delete pvc example-vanilla-block-pvc
+persistentvolumeclaim "example-vanilla-block-pvc" deleted
+```
+
+The PV object from the cluster also gets deleted
+
+```shell
+# kubectl get pv pvc-6791fdd4-5fad-438e-a7fb-16410363e3da
+Error from server (NotFound): persistentvolumes "pvc-6791fdd4-5fad-438e-a7fb-16410363e3da" not found
+```
+
+However, the underlying storage resource is not deleted, and needs to be removed manually.
+
+To sum it up, Reclaim policy associated with the Persistent Volume is currently ignored under certain circumstance. For a `Bound` PV-PVC pair the ordering of PV-PVC deletion determines whether the PV delete reclaim policy is honored. The PV honors the reclaim policy if the PVC is deleted, however, if the PV is deleted prior to deleting the PVC then the Reclaim policy is not exercised. As a result of this behavior, the associated storage asset in the external infrastructure is not removed.
+
+## What is the new behavior?
+
+The new behavior ensures that the underlying storage object is deleted from the backend when users attempt to delete a PV manually.
+
+The fix is introduced as an alpha in 1.23 under the feature-gate `HonorPVReclaimPolicy`. Update the `external-provisioner` to the `v4.0.0` and enable the feature gate.
```

Comment:
**This (the fact it's an alpha feature and off by default) needs highlighting really prominently.**